src/ops/conversion.rs

```diff
@@ -0,0 +1,244 @@
+use mlx_macros::default_device;
+
+use crate::{Array, ArrayElement, StreamOrDevice};
+
+impl Array {
+    /// Cast the array to a specified type.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::Array;
+    ///
+    /// let array = Array::from_slice(&[1i16,2,3], &[3]);
+    /// let mut new_array = array.as_type::<f32>();
+    /// new_array.eval();
+    ///
+    /// assert_eq!(new_array.dtype(), mlx::Dtype::Float32);
+    /// assert_eq!(new_array.shape(), &[3]);
+    /// assert_eq!(new_array.item_size(), 4);
+    /// assert_eq!(new_array.as_slice::<f32>(), &[1.0,2.0,3.0]);
```

Comment:
are we doing asserts in the examples after all?

Reply:
I don't think we have to. I just felt for this function it's a better way to show what remains the same what's changed in the type conversion

Reply:
Makes sense!